RELEASE_NOTES.md

```diff
@@ -0,0 +1,57 @@
+# 3.4 
+
+The top four improvements in Dafny 3.4 are:
```

Comment:
I don't think we need this line.

Reply:
I'm inclined to agree. I modelled this on the 3.3 release notes that do include a summary section, although they're significantly bigger in total: https://github.com/dafny-lang/dafny/releases/tag/v3.3.0

Still think we should leave it out?

Reply:
I like the headline features, but I don't think they need an introduction.

Reply:
I definitely like highlighting the features at least.  I don't have a strong opinion about this exact line.